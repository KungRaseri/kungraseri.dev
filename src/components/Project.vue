
<template>
  <v-flex md4 xm2>
    <v-container>
      <v-card
        :id="`project-${project.id}`"
        class="mx-auto"
        :flat="flat"
        :loading="loading"
        :outlined="outlined"
        :elevation="elevation"
        :raised="raised"
        :width="width"
        :height="height"
      >
        <v-img v-if="media" class="white--text" height="200px" :src="project.url">
          <v-card-title class="align-end fill-height">{{project.imageTitle}}</v-card-title>
        </v-img>
        <v-card-title v-else>{{project.title}}</v-card-title>

        <v-card-text>{{project.description}}</v-card-text>
        <v-card-actions v-if="actions">
          <v-btn icon>
            <v-icon>mdi-heart</v-icon>
          </v-btn>
          <v-btn text>Click</v-btn>
          <v-btn outlined>Click</v-btn>
        </v-card-actions>
      </v-card>
    </v-container>
  </v-flex>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';

@Component
export default class Project extends Vue {
  @Prop({ type: Object })
  public project: object | undefined;
}
</script>
