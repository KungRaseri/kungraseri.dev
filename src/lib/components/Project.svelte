<script lang="ts">
	import type { Project } from '$lib/types';

	export let project: Project;
</script>

<div class="card card-hover">
	<a href={project.url}>
		<img class="bg-black/50 w-full aspect-video" src={project.image} alt={project.title} />
	</a>
	<header class="card-header">
		<h1 class="h2 sm:h1">{project.title}</h1>
	</header>
	<section class="p-4">{project.description}</section>

	<footer class="card-footer">
		<div class="btn-group flex justify-center gap-1">
			{#each project.actions as action}
				<a href={action.href} class="btn btn-sm variant-soft-secondary">{action.text}</a>
			{/each}
		</div>
	</footer>
</div>
