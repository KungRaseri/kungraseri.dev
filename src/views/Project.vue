
<template>
<div class="project">
  project
</div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component({
  components: {
  },
})
export default class Project extends Vue {}
</script>
