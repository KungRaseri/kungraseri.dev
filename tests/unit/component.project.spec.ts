import Vue from 'vue';
import Vuetify from 'vuetify';

import { mount, createLocalVue } from '@vue/test-utils';

import Project from '@/components/Project.vue';

const localVue = createLocalVue();
localVue.use(Vuetify);

describe('Project.vue', () => {
  test('renders props when passed', () => {
    const project = {
      id: 0,
      title: 'Test Title',
      image: 'test.png',
      url: '#',
    };
    const wrapper = mount(Project, {
      propsData: { project },
    });

    expect(wrapper.props).toHaveProperty(project.title);
  });
});
