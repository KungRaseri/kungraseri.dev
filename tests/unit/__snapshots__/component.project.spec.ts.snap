// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project.vue renders props when passed 1`] = `
<div class="flex md4 xm2">
  <div class="container">
    <div class="mx-auto v-card v-sheet theme--light" id="project-0">
      <div class="v-card__title">Test Title</div>
      <div class="v-card__text">Test Description</div>
      <!---->
    </div>
  </div>
</div>
`;
